src/app/shared/pokemon.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@angular/core';
+import { Pokemon } from './pokemon';
+import { HttpClient } from '@angular/common/http';
+import { Observable, forkJoin, map, mergeMap } from 'rxjs';
+
+
+const URL = 'https://pokeapi.co/api/v2/pokemon/?offset=0&limit=100/';
+  
+@Injectable({
+  providedIn: 'root'
+})
+export class PokemonService {
+  pokemons: Pokemon[];
+  clickedPokemon: Pokemon;
```

Comment:
You can to know what is the selected pokemon by the param in route. You don't need to save it here